R/Lrnr_hal9001.R

```diff
@@ -73,7 +73,28 @@ Lrnr_hal9001 <- R6Class(
       if (!any(grepl("fit_control", names(args)))) {
         args$fit_control <- list()
       }
-      args$fit_control$foldid <- origami::folds2foldvec(task$folds)
+
+      if (!any(grepl("fold", names(args$fit_control)))) {
+        args$fit_control$foldid <- origami::folds2foldvec(task$folds)
+      } else {
```

Comment:
@rachaelvp Could we clarify this part of the logic with some comments? I'm having trouble following exactly what happens here with stratifying on `Y` given the several conditional blocks